### PR TITLE
Music background #12

### LINKED ISSUE #12: music background
https://stackoverflow.com/questions/6529645/how-to-add-background-music-to-a-web-page

### PR DESCRIPTION
**Pull request by aleksandraj036**

### CODE REVIEW

File: src/app/utils/audio.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const music = document.addEventListener("DOMContentLoaded", event => {
+    const audio = document.querySelector("audio");
+    audio.volume = 0.7;
```
Comment (hoolek77) [inline src/app/utils/audio.ts]: There's an error. You don't specify type of audio variable. You can fix it by adding 'as HTMLAudioElement' at the end of line 2.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5501
+}
```
Comment (hoolek77) [inline .vscode/settings.json]: You should remove this file and directory and next time start the application by command 'npm start'

**Review by hoolek77 — CHANGES_REQUESTED** [review]

File: src/app/index.ts [hunk 3]
```diff
@@ -1,2 +1,6 @@
 export * from './app'
+aj-12
+export * from './utils/audio'
```
Comment (pdybowski) [inline src/app/index.ts]: Change it just to './utils'
In utils/index.ts add export * from './audio'

File: src/app/index.ts [hunk 4]
```diff
@@ -1,2 +1,6 @@
 export * from './app'
+aj-12
```
Comment (pdybowski) [inline src/app/index.ts]: It will cause an error. Probably undefined variable

File: src/app/index.ts [hunk 5]
```diff
@@ -1,2 +1,6 @@
...
+export * from './utils/audio'
+
 export * from './components'
+master
```
Comment (pdybowski) [inline src/app/index.ts]: It will cause an error

File: src/app/utils/audio.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export const music = document.addEventListener("DOMContentLoaded", event => {
```
Comment (pdybowski) [inline src/app/utils/audio.ts]: Okay, this one is bigger. Change it to a function - playAudio(volume) and import it in app.ts file. Run this method in App.start(). Volume argument should be used to init vaudio volume. Let's say default value in app.ts should be set to 0.7
Remove event listener in that file. App.start() is triggered inside this event. Moreover you can add new method in audio.ts file such as audioSettings(volume). Then in the body of this function run playAudio(volume)

File: src/app/utils/audio.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+    const audio = document.querySelector("audio") as HTMLAudioElement;
+    audio.volume = 0.7;
```
Comment (pdybowski) [inline src/app/utils/audio.ts]: Audio.volume = volume ?? 0.7

**Review by pdybowski — CHANGES_REQUESTED** [review]
Small changes

File: src/app/app.ts [hunk 8]
```diff
@@ -1,5 +1,7 @@
+import {audioSettings} from './utils'
 export class App {
   start() {
     console.log('Starting application...')
   }
+  audioSettings() {};
```
Comment (pdybowski) [inline src/app/app.ts]: Remove it for now. There will be Better place for it

File: webpack.common.js [hunk 9]
```diff
@@ -38,6 +38,14 @@ module.exports = {
           },
         ],
       },
+      {
+        test: /\.mp3$/,
```
Comment (pdybowski) [inline webpack.common.js]: Move this regex to test where png,jpg etc are 
You don't need to specify the same loader twice

**Review by pdybowski — CHANGES_REQUESTED** [review]
Jeszcze cos 🤓

File: src/app/utils/audio.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export const playAudio = function(volume: number) {
```
Comment (pdybowski) [inline src/app/utils/audio.ts:1]: Run this function inside App Class, inside start() method. Below console.log(Starting application...),
Set argument value to 0.7

File: src/app/app.ts [hunk 11]
```diff
@@ -1,3 +1,4 @@
+import {audioSettings} from './utils'
```
Comment (pdybowski) [inline src/app/app.ts]: nah, sent you image on discord

**Review by pdybowski — CHANGES_REQUESTED** [review]

**Review by pdybowski — APPROVED** [review]

**Review by hoolek77 — APPROVED** [review]

**Review by bartoszbialecki — APPROVED** [review]

**Pull request merged**